File: stringSplosion.java
/*
Given a non-empty string like "Code" return a string like "CCoCodCode".


stringSplosion("Code") → "CCoCodCode"
stringSplosion("abc") → "aababc"
stringSplosion("ab") → "aab"
*/

public String stringSplosion(String str){
  String splosion = "";
  for(int i = 0; i < str.length(); i++){
    splosion += str.substring(0, i);
  }
  return splosion += str;
}

File: Python/Warmup-2/string_splosion.py
# Given a non-empty string like "Code" return a string like "CCoCodCode".

def string_splosion(str):
  splosion = ""
  
  # len(str) + 1 so that we can reach the last character of the string
  for i in range(len(str) + 1):
    splosion += str[:i]
  return splosion
